flaky test: `chainerx_tests/unit_tests/routines_tests/test_arithmetic.py::test_Sub`

https://jenkins.preferred.jp/job/chainer/job/chainer_pr/1709/TEST=CHAINERX_chainerx-py3,label=mn1-p100/console
```
14:13:12 FAIL ../../../repo/tests/chainerx_tests/unit_tests/routines_tests/test_arithmetic.py::test_Sub_param_133_{in_dtypes=('float16', 'float16'), out_dtype='float16', in_shapes=((3, 1), (2, 1, 4)), input_lhs='random', input_rhs='random', is_module=False}[cuda:0]
```
```
14:13:10 _ test_Sub_param_133_{in_dtypes=('float16', 'float16'), out_dtype='float16', in_shapes=((3, 1), (2, 1, 4)), input_lhs='random', input_rhs='random', is_module=False}[cuda:0] _
14:13:10 
14:13:10 device = cuda:0, args = (), kwargs = {}
14:13:10 backend_config = <BackendConfig use_chainerx=True chainerx_device='cuda:0' use_cuda=False cuda_device=None use_cudnn='never' cudnn_deterministic=False autotune=False cudnn_fast_batch_normalization=False use_ideep='never'>
14:13:10 obj = <chainer.testing._bundle.TestSub_param_133_{in_dtypes=('float16', 'float16'), out_dtype='float16', in_shapes=((3, 1), (2, 1, 4)), input_lhs='random', input_rhs='random', is_module=False} object at 0x7fcfe4285eb8>
14:13:10 
14:13:10     @pytest.mark.parametrize_device(devices)
14:13:10     def entry_func(device, *args, **kwargs):
14:13:10         backend_config = _make_backend_config(device.name)
14:13:10     
14:13:10         # Forward test
14:13:10         obj = cls()
14:13:10         try:
14:13:10             obj.setup(*args, **kwargs)
14:13:10             obj.run_test_forward(backend_config)
14:13:10         finally:
14:13:10             obj.teardown()
14:13:10     
14:13:10         # If this is a NumpyOpTest instance, skip backward/double-backward
14:13:10         # tests if the forward test succeeds with acceptable errors.
14:13:10         if isinstance(obj, NumpyOpTest):
14:13:10             if obj.is_forward_successful_with_accept_errors:
14:13:10                 return  # success with expected errors
14:13:10     
14:13:10         # Backward test
14:13:10         obj = cls()
14:13:10         try:
14:13:10             obj.setup(*args, **kwargs)
14:13:10 >           obj.run_test_backward(backend_config)
14:13:10 
14:13:10 args       = ()
14:13:10 backend_config = <BackendConfig use_chainerx=True chainerx_device='cuda:0' use_cuda=False cuda_device=None use_cudnn='never' cudnn_deterministic=False autotune=False cudnn_fast_batch_normalization=False use_ideep='never'>
14:13:10 cls        = <class 'chainer.testing._bundle.TestSub_param_133_{in_dtypes=('float16', 'float16'), out_dtype='float16', in_shapes=((3, 1), (2, 1, 4)), input_lhs='random', input_rhs='random', is_module=False}'>
14:13:10 device     = cuda:0
14:13:10 kwargs     = {}
14:13:10 obj        = <chainer.testing._bundle.TestSub_param_133_{in_dtypes=('float16', 'float16'), out_dtype='float16', in_shapes=((3, 1), (2, 1, 4)), input_lhs='random', input_rhs='random', is_module=False} object at 0x7fcfe4285eb8>
14:13:10 
14:13:10 /repo/tests/chainerx_tests/op_utils.py:366: 
14:13:10 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
14:13:10 /repo/tests/chainerx_tests/op_utils.py:95: in run_test_backward
14:13:10     super(OpTest, self).run_test_backward(backend_config)
14:13:10 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/testing/function_link.py:254: in run_test_backward
14:13:10     do_check()
14:13:10 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/testing/function_link.py:243: in do_check
14:13:10     **self.check_backward_options)
14:13:10 /workspace/conda/envs/testenv/lib/python3.6/contextlib.py:99: in __exit__
14:13:10     self.gen.throw(type, value, traceback)
14:13:10 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/testing/function_link.py:36: in raise_if_fail
14:13:10     cls.fail(message, e)
14:13:10 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/testing/function_link.py:27: in fail
14:13:10     utils._raise_from(cls, message, exc)
14:13:10 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/utils/__init__.py:106: in _raise_from
14:13:10     six.raise_from(new_exc.with_traceback(orig_exc.__traceback__), None)
14:13:10 <string>:3: in raise_from
14:13:10     ???
14:13:10 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/testing/function_link.py:34: in raise_if_fail
14:13:10     yield
14:13:10 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/testing/function_link.py:243: in do_check
14:13:10     **self.check_backward_options)
14:13:10 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/gradient_check.py:900: in check_backward
14:13:10     detect_nondifferentiable, is_immutable_params=False
14:13:10 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/gradient_check.py:464: in run
14:13:10     self._run()
14:13:10 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/gradient_check.py:507: in _run
14:13:10     self._compare_gradients(gx_numeric, gx_backward, directions)
14:13:10 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
14:13:10 
14:13:10 self = <chainer.gradient_check._CheckBackward object at 0x7fcfe4285668>
14:13:10 gx_numeric = array(-0.12077938, shape=(), dtype=float64, device='cuda:0')
14:13:10 gx_backward = array(-0.11963368, shape=(), dtype=float64, device='cuda:0')
14:13:10 directions = [array([[-0.535809  ],
14:13:10        [-0.38725089],
14:13:10        [-0.1564945 ]], shape=(3, 1), dtype=float64, device='cuda:0'), arr...896]],
14:13:10 
14:13:10        [[0.17207419, 0.0685811 , -0.10751192, -0.09257840]]], shape=(2, 1, 4), dtype=float64, device='cuda:0')]
14:13:10 
14:13:10     def _compare_gradients(self, gx_numeric, gx_backward, directions):
14:13:10         atol = self.atol
14:13:10         rtol = self.rtol
14:13:10         # Compare the gradients
14:13:10         try:
14:13:10             testing.assert_allclose(
14:13:10                 gx_numeric, gx_backward, atol=atol, rtol=rtol)
14:13:10         except AssertionError as e:
14:13:10             eps = self.eps
14:13:10             xs = self.xs
14:13:10             gys = self.gys
14:13:10             f = six.StringIO()
14:13:10             f.write('check_backward failed (eps={} atol={} rtol={})\n'.format(
14:13:10                 eps, atol, rtol))
14:13:10             for i, x in enumerate(xs):
14:13:10                 f.write('inputs[{}]:\n'.format(i))
14:13:10                 f.write('{}\n'.format(x))
14:13:10             for i, gy in enumerate(gys):
14:13:10                 f.write('grad_outputs[{}]:\n'.format(i))
14:13:10                 f.write('{}\n'.format(gy))
14:13:10             for i, d in enumerate(directions):
14:13:10                 f.write('directions[{}]:\n'.format(i))
14:13:10                 f.write('{}\n'.format(d))
14:13:10             f.write('gradients (numeric):  {}\n'.format(gx_numeric))
14:13:10             f.write('gradients (backward): {}\n'.format(gx_backward))
14:13:10             f.write('\n')
14:13:10             f.write('x: numeric gradient, y: backward gradient')
14:13:10             f.write(str(e))
14:13:10 >           raise AssertionError(f.getvalue())
14:13:10 E           chainer.testing.function_link.FunctionTestError: backward is not implemented correctly
14:13:10 E           
14:13:10 E           (caused by)
14:13:10 E           AssertionError: check_backward failed (eps=0.001 atol=0.001 rtol=0.001)
14:13:10 E           inputs[0]:
14:13:10 E           array([[0.94433594],
14:13:10 E                  [0.60546875],
14:13:10 E                  [0.60449219]], shape=(3, 1), dtype=float16, device='cuda:0')
14:13:10 E           inputs[1]:
14:13:10 E           array([[[-0.5727539 , 0.41992188, -0.49975586, 0.01114655]],
14:13:10 E           
14:13:10 E                  [[-0.42333984, 0.36157227, 0.0670166 , -0.98388672]]], shape=(2, 1, 4), dtype=float16, device='cuda:0')
14:13:10 E           grad_outputs[0]:
14:13:10 E           array([[[0.18603516, -0.89111328, 0.93115234, -0.79003906],
14:13:10 E                   [0.30664062, 0.88183594, -0.00023222, 0.76123047],
14:13:10 E                   [-0.39160156, -0.74902344, 0.13085938, 0.8305664 ]],
14:13:10 E           
14:13:10 E                  [[-0.16918945, 0.92822266, 0.26513672, -0.79785156],
14:13:10 E                   [0.0625    , 0.19274902, 0.0118637 , 0.98339844],
14:13:10 E                   [0.69970703, -0.92773438, -0.79785156, -0.75683594]]], shape=(2, 3, 4), dtype=float16, device='cuda:0')
14:13:10 E           directions[0]:
14:13:10 E           array([[-0.535809  ],
14:13:10 E                  [-0.38725089],
14:13:10 E                  [-0.1564945 ]], shape=(3, 1), dtype=float64, device='cuda:0')
14:13:10 E           directions[1]:
14:13:10 E           array([[[0.18233666, 0.55440092, -0.37171367, -0.07234896]],
14:13:10 E           
14:13:10 E                  [[0.17207419, 0.0685811 , -0.10751192, -0.09257840]]], shape=(2, 1, 4), dtype=float64, device='cuda:0')
14:13:10 E           gradients (numeric):  array(-0.12077938, shape=(), dtype=float64, device='cuda:0')
14:13:10 E           gradients (backward): array(-0.11963368, shape=(), dtype=float64, device='cuda:0')
14:13:10 E           
14:13:10 E           x: numeric gradient, y: backward gradient
14:13:10 E           Not equal to tolerance rtol=0.001, atol=0.001
14:13:10 E           
14:13:10 E           Mismatch: 100%
14:13:10 E           Max absolute difference: 0.0011457
14:13:10 E           Max relative difference: 0.00957673
14:13:10 E            x: array(-0.120779)
14:13:10 E            y: array(-0.119634)
14:13:10 E           
14:13:10 E           assert_allclose failed: 
14:13:10 E             shape: () ()
14:13:10 E             dtype: float64 float64
14:13:10 E             i: (0,)
14:13:10 E             x[i]: -0.12077937973014614
14:13:10 E             y[i]: -0.11963368028302046
14:13:10 E             relative error[i]: 0.009576729934373607
14:13:10 E             absolute error[i]: 0.0011456994471256837
14:13:10 E           x: -0.12077938
14:13:10 E           y: -0.11963368
14:13:10 
14:13:10 atol       = 0.001
14:13:10 d          = array([[[0.18233666, 0.55440092, -0.37171367, -0.07234896]],
14:13:10 
14:13:10        [[0.17207419, 0.0685811 , -0.10751192, -0.09257840]]], shape=(2, 1, 4), dtype=float64, device='cuda:0')
14:13:10 directions = [array([[-0.535809  ],
14:13:10        [-0.38725089],
14:13:10        [-0.1564945 ]], shape=(3, 1), dtype=float64, device='cuda:0'), arr...896]],
14:13:10 
14:13:10        [[0.17207419, 0.0685811 , -0.10751192, -0.09257840]]], shape=(2, 1, 4), dtype=float64, device='cuda:0')]
14:13:10 eps        = 0.001
14:13:10 f          = <_io.StringIO object at 0x7fcfe420c048>
14:13:10 gx_backward = array(-0.11963368, shape=(), dtype=float64, device='cuda:0')
14:13:10 gx_numeric = array(-0.12077938, shape=(), dtype=float64, device='cuda:0')
14:13:10 gy         = array([[[0.18603516, -0.89111328, 0.93115234, -0.79003906],
14:13:10         [0.30664062, 0.88183594, -0.00023222, 0.76123047],...39844],
14:13:10         [0.69970703, -0.92773438, -0.79785156, -0.75683594]]], shape=(2, 3, 4), dtype=float16, device='cuda:0')
14:13:10 gys        = (array([[[0.18603516, -0.89111328, 0.93115234, -0.79003906],
14:13:10         [0.30664062, 0.88183594, -0.00023222, 0.76123047]...844],
14:13:10         [0.69970703, -0.92773438, -0.79785156, -0.75683594]]], shape=(2, 3, 4), dtype=float16, device='cuda:0'),)
14:13:10 i          = 1
14:13:10 rtol       = 0.001
14:13:10 self       = <chainer.gradient_check._CheckBackward object at 0x7fcfe4285668>
14:13:10 x          = array([[[-0.5727539 , 0.41992188, -0.49975586, 0.01114655]],
14:13:10 
14:13:10        [[-0.42333984, 0.36157227, 0.0670166 , -0.98388672]]], shape=(2, 1, 4), dtype=float16, device='cuda:0')
14:13:10 xs         = (array([[0.94433594],
14:13:10        [0.60546875],
14:13:10        [0.60449219]], shape=(3, 1), dtype=float16, device='cuda:0'), array(...655]],
14:13:10 
14:13:10        [[-0.42333984, 0.36157227, 0.0670166 , -0.98388672]]], shape=(2, 1, 4), dtype=float16, device='cuda:0'))
14:13:10 
14:13:10 /workspace/conda/envs/testenv/lib/python3.6/site-packages/chainer/gradient_check.py:537: FunctionTestError
```

--- a/tests/chainerx_tests/math_utils.py
+++ b/tests/chainerx_tests/math_utils.py
@@ -77,9 +77,9 @@ class BinaryMathTestBase(object):
 
         if in_dtype1 == 'float16' or in_dtype2 == 'float16':
             self.check_forward_options.update({'rtol': 1e-3, 'atol': 1e-3})
-            self.check_backward_options.update({'rtol': 1e-3, 'atol': 1e-3})
+            self.check_backward_options.update({'rtol': 1e-2, 'atol': 3e-3})
             self.check_double_backward_options.update(
-                {'rtol': 1e-3, 'atol': 1e-3})
+                {'rtol': 1e-2, 'atol': 3e-3})
 
     def generate_inputs(self):
         in_dtype1, in_dtype2 = self.in_dtypes

